addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -5250,51 +5279,99 @@ registry.layout_column = SnippetOptionWidget.extend({
                 });
             }
-        } else {
-            var self = this;
-            for (const el of $row.children().slice(count)) {
+        }
+        // Expected behavior: remove items if they are already all on one row.
+        // Otherwise, keep all items and just reorganize the columns on the
+        // right amount of rows.
+        else if (itemsDelta < 0 && prevNbRows < 2) {
+            for (const el of [...rowEl.children].slice(itemsDelta)) {
                 await new Promise(resolve => {
-                    self.trigger_up('remove_snippet', {$snippet: $(el), onSuccess: resolve, shouldRecordUndo: false});
+                    this.trigger_up('remove_snippet', {
+                        $snippet: $(el),
+                        onSuccess: resolve,
+                        shouldRecordUndo: false,
+                    });
                 });
             }
         }
 
-        this._resizeColumns($row.children());
+        this._resizeColumns(rowEl.children, nbColumns);
         this.trigger_up('cover_update');
     },
     /**
-     * Resizes the columns so that they are kept on one row.
+     * Resizes the columns according to mobile or desktop view.
      *
      * @private
-     * @param {jQuery} $columns - the columns to resize
-     */
-    _resizeColumns: function ($columns) {
-        const colsLength = $columns.length;
-        var colSize = Math.floor(12 / colsLength) || 1;
-        var colOffset = Math.floor((12 - colSize * colsLength) / 2);
-        var colClass = 'col-lg-' + colSize;
-        $columns.toArray().forEach((column) => {
-            var $column = $(column);
-            $column.attr('class', $column.attr('class').replace(/\b(col|offset)-lg(-\d+)?\b/g, ''));
-            $column.addClass(colClass);
+     * @param {HTMLCollection} columnEls - the elements to resize
+     * @param {integer} nbColumns - the number of wanted columns
+     */
+    _resizeColumns(columnEls, nbColumns) {
+        const isMobile = weUtils.isMobile(this);
+        const sameNbOfColsMobileDesktop = nbColumns === this._getNbColumns(!isMobile);
+        const resolutionModifier = isMobile ? '' : '-lg';
+        const itemSize = Math.floor(12 / nbColumns) || 1;
+        const firstItemOffset = this._getItemOffset(itemSize, nbColumns);
+        let resolutionRegex;
+        let resetMobileItemSize;
+
+        if (sameNbOfColsMobileDesktop) {
+            resolutionRegex = /\b(col|offset)(-lg)?(-\d{1,2})?(?![\w-])/g;
+        } else {
+            resolutionRegex =
+                new RegExp(`\\b(col|offset)(${resolutionModifier})?(-\\d{1,2})?(?![\\w-])`,'g');
+
+            // When the desktop view changes (adds or removes items), the
+            // mobile view is reset accordingly.
+            if (!isMobile) {
+                const mobileColumns = this._getNbColumns(true);
+                resetMobileItemSize = Math.floor(12 / mobileColumns) || 1;
+            }
+        }
+
+        [...columnEls].forEach(columnEl => {
```

Comment:
Especially when you have to do `[...myStuffWhichIsNotReallyAnArray]`, I think `for (const columnEl of columnEls) {` is a prettier syntax (first the temporary variable can be a `const` but it also can be "read" naturally "Pour chaque column parmi ces columns, do...". Feel free to disagree, if you prefer that way, it's ok :wink: